src/visualization/mapContainer.tsx

```diff
@@ -65,13 +74,20 @@ const MapContainer = ({ location }: RouteComponentProps) => {
           setViewport(vp);
         }}
       >
-        {queryParams && (
-          <Marker
-            latitude={queryParams.lat}
-            longitude={queryParams.lon}
-            className="mapboxgl-user-location-dot"
-          />
-        )}
+        {queryParams &&
+          (isOnline ? (
+            <Marker
+              latitude={queryParams.lat}
+              longitude={queryParams.lon}
+              className="mapboxgl-user-location-dot"
+            />
+          ) : (
+            <OfflineMarker
+              latitude={queryParams.lat}
+              longitude={queryParams.lon}
+              className="mapboxgl-user-location-dot"
+            />
```

Comment:
There is a bit duplication in lines 79-89, we could reduce the duplication by something like this:

in render func beginnning:
`const UserMarker = isOnline ? Marker : OfflineMarker`

and then just use the `UserMarker` component.